Module not found: Error: Can't resolve 'use-context-selector'

# Bug Reports

### Checklist

- [x] I have read through the [FAQ](https://formidable.com/open-source/victory/docs/faq) and [Guides](https://formidable.com/open-source/victory/guides/)

- [x] I am using the latest version of Victory

- [x] I've searched open issues to make sure I'm not opening a duplicate issue

### The Problem

victory-core/es/victory-state seems recently to have added a dependence on `use-context-selector` without expressing this dependence in its package.json. As a result, webpack builds now fail.

Further complicating matters... the latest several versions of `use-context-selector` depend on `react >= 18`... so I'm not even sure how you all would express a dependence on this node module in a way that allowed for that kind of flexibility. oof. 

of note, patternfly, which uses you, is not yet react 18, and yet they express a `^` floating dependence on you that brings in this breaking change!

Full error message:
> Module not found: Error: Can't resolve 'use-context-selector' in '.../node_modules/victory-core/es/victory-state'

The fix, for react 17 users, is `npm install use-context-selector@1.1`; >1.1 have a react >= 18 dependence


### Reproduction

todo

--- a/package.json
+++ b/package.json
@@ -120,7 +120,6 @@
     "ts-jest": "^28.0.3",
     "ts-loader": "^9.3.0",
     "typescript": "^4.7.3",
-    "use-context-selector": "^1.4.1",
     "webpack": "^5.72.1",
     "webpack-cli": "^4.9.2",
     "webpack-dev-server": "^4.9.0",

--- a/packages/victory-core/src/victory-state/victory-provider.tsx
+++ b/packages/victory-core/src/victory-state/victory-provider.tsx
@@ -1,5 +1,5 @@
 import * as React from "react";
-import { createContext, useContextSelector } from "use-context-selector";
+import { createContext, useContext } from "react";
 import { D3ScaleFn, DomainTuple, ForAxes } from "../types/prop-types";
 import { FormattedDatum, getData } from "./helpers/get-data";
 import { getDomain } from "./helpers/get-domain";
@@ -81,17 +81,12 @@ export function VictoryProvider({
   );
 }
 
-type ContextValue = ContextType | null;
-
 export function useVictoryContext<T>(selector: (value: ContextType) => T): T {
-  return useContextSelector<ContextValue, T>(VictoryContext, (context) => {
-    if (!context) {
-      throw new Error(
-        "useVictoryContext must be used within a VictoryProvider",
-      );
-    }
-    return selector(context);
-  });
+  const context = useContext(VictoryContext);
+  if (!context) {
+    throw new Error("useVictoryContext must be used within a VictoryProvider");
+  }
+  return selector(context);
 }
 
 export function useScale() {

